From 8bb63f77bb5fa46d2749ec4d3662ac88b7ff5938 Mon Sep 17 00:00:00 2001
From: Markus Armbruster <armbru@redhat.com>
Date: Wed, 18 Jan 2012 10:38:36 +0100
Subject: [PATCH 42/52] raw-posix: Always check paio_init result

RH-Author: Markus Armbruster <armbru@redhat.com>
Message-id: <1326883126-22053-43-git-send-email-armbru@redhat.com>
Patchwork-id: 36609
O-Subject: [RHEL-6.3 PATCH qemu-kvm 42/52] raw-posix: Always check paio_init result
Bugzilla: 758194
RH-Acked-by: Miroslav Rezanina <mrezanin@redhat.com>
RH-Acked-by: Laszlo Ersek <lersek@redhat.com>
RH-Acked-by: Amit Shah <amit.shah@redhat.com>

From: Frediano Ziglio <freddy77@gmail.com>

Signed-off-by: Frediano Ziglio <freddy77@gmail.com>
Signed-off-by: Kevin Wolf <kwolf@redhat.com>
(cherry picked from commit f6e8ffc22fe153ba981f2747e4c52ea7e55f6ecc)
Signed-off-by: Markus Armbruster <armbru@redhat.com>
---
 block/raw-posix.c |   11 +++++------
 1 files changed, 5 insertions(+), 6 deletions(-)

Signed-off-by: Michal Novotny <minovotn@redhat.com>
---
 block/raw-posix.c |   11 +++++------
 1 files changed, 5 insertions(+), 6 deletions(-)

diff --git a/block/raw-posix.c b/block/raw-posix.c
index cc4bc76..791199e 100644
--- a/block/raw-posix.c
+++ b/block/raw-posix.c
@@ -211,13 +211,15 @@ static int raw_open_common(BlockDriverState *bs, const char *filename,
         }
     }
 
+    /* We're falling back to POSIX AIO in some cases so init always */
+    if (paio_init() < 0) {
+        goto out_free_buf;
+    }
+
 #ifdef CONFIG_LINUX_AIO
     if ((bdrv_flags & (BDRV_O_NOCACHE|BDRV_O_NATIVE_AIO)) ==
                       (BDRV_O_NOCACHE|BDRV_O_NATIVE_AIO)) {
 
-        /* We're falling back to POSIX AIO in some cases */
-        paio_init();
-
         s->aio_ctx = laio_init();
         if (!s->aio_ctx) {
             goto out_free_buf;
@@ -226,9 +228,6 @@ static int raw_open_common(BlockDriverState *bs, const char *filename,
     } else
 #endif
     {
-        if (paio_init() < 0) {
-            goto out_free_buf;
-        }
 #ifdef CONFIG_LINUX_AIO
         s->use_aio = 0;
 #endif
-- 
1.7.7.5

